internal/controller/compute/cubeserver/cubeserver.go

```
@@ -224,10 +228,28 @@ func (c *externalServer) Update(ctx context.Context, mg resource.Managed) (manag
 		return managed.ExternalUpdate{}, errors.New(errNotCubeServer)
 	}
 	if cr.Status.AtProvider.State == compute.BUSY || cr.Status.AtProvider.State == compute.UPDATING {
+		c.log.Debug(fmt.Sprintf("Update, cube server state is %s, waiting...", cr.Status.AtProvider.State), "name", cr.Spec.ForProvider.Name)
 		return managed.ExternalUpdate{}, nil
 	}
 
 	serverID := cr.Status.AtProvider.ServerID
+
+	// Handle RUNNING state before update - you cannot update a cube server in a SUSPENDED state
```

in the case there is an attempt to update a suspended cube without the update also triggering an unsuspend, should we fail early here, before the api request is made ?